core/src/main/java/org/wso2/msf4j/Interceptor.java

```diff
@@ -16,10 +16,18 @@
 
 package org.wso2.msf4j;
 
+import org.wso2.msf4j.interceptor.MSF4JRequestInterceptor;
+import org.wso2.msf4j.interceptor.MSF4JResponseInterceptor;
+import org.wso2.msf4j.internal.MSF4JConstants;
+
+import java.lang.reflect.Method;
+
 /**
  * Interface that needs to be implemented to intercept handler method calls.
+ *
+ * @deprecated
  */
-public interface Interceptor {
+public interface Interceptor extends MSF4JRequestInterceptor, MSF4JResponseInterceptor {
```

Comment:
The interceptors (deprecated) will work as before without any difference. However, internally they function as global request and global response interceptors. The idea is to avoid code smells. This was suggested in the code review we had.